### PR TITLE
planner, executor: let 'show [variables|status]' be sorted by 'Variable_name'

### PR DESCRIPTION
**Pull request by rebelice**
### What problem does this PR solve? 
https://github.com/pingcap/tidb/issues/14542
Let 'show [variables|status]' be sorted by 'Variable_name'. And I add tests for it.

### What is changed and how it works?
I add a logical sort for 'show [variables|status]'.

### Check List 

Tests 

 - Unit test
 - Integration test

Code changes

 - Has exported function/method change

Side effects

Related changes

Release note

### CODE REVIEW

File: planner/core/planbuilder.go [hunk 1]
```diff
@@ -1644,9 +1644,19 @@ func (b *PlanBuilder) buildShow(ctx context.Context, show *ast.ShowStmt) (Plan,
...
-	return p, nil
+		np = proj
+	}
+	if show.Tp == ast.ShowVariables || show.Tp == ast.ShowStatus {
+		b.curClause = orderByClause
+		sort := LogicalSort{}.Init(b.ctx, b.getSelectOffset())
+		exprs := make([]*ByItems, 0, 1)
+		newCol := np.Schema().Columns[0].Clone().(*expression.Column)
+		exprs = append(exprs, &ByItems{Expr: newCol, Desc: false})
+		sort.ByItems = exprs
```
Comment (SunRunAway) [inline planner/core/planbuilder.go]:
```
		newCol := np.Schema().Columns[0].Clone().(*expression.Column)
		sort := LogicalSort{
			ByItems: []*ByItems{{Expr: newCol}},
		}.Init(b.ctx, b.getSelectOffset())
```

Comment (rebelice) [reply planner/core/planbuilder.go]: Looks great. I'll modify it right away.

**Pull request merged**